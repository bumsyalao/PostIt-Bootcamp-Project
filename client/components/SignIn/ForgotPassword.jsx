import React, { Component } from 'react';
import { connect } from 'react-redux';
import { forgotPasswordAction } from '../../actions/SignInAction';
import NavigationBar from '../NavigationBar';

class ForgotPassword extends Component {
  constructor(props){
    super(props);
    this.state = {
      email: ''
    }
    this.onChange = this.onChange.bind(this);
    this.forgotPassword = this.forgotPassword.bind(this);
  }
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }
  forgotPassword(event)  {
    const email = this.state.email;
    this.props.forgotPasswordAction({email})
      .then(() => {
          this.props.history.push('/');
          Materialize.toast('Password Reset Link sent', 5000, 'green');

      })
      .catch(error => {
        console.log(error);
        Materialize.toast('Server Error', 5000, 'red');
      })
  }
  render(){

    return (
      <div>
      <NavigationBar match={this.props.match}/>
      <div className="form-margin">
        <div className="col s12 container">
          <div className="input-field col s6 offset-s3">
            <i className="material-icons prefix">email</i>
            <input
              id="email"
              value={this.state.email}
              onChange={this.onChange}
              name="email"
              type="text"
              className="validate"
              required
            />
            <label className="active" htmlFor="credential">
              Type in your email address so we can send you instructions to reset your password
            </label>
          </div>
          <div className="input-field col s6 offset-s3">
            <button
              onClick={this.forgotPassword}
              className="btn waves-effect waves-light col s6 offset-s3 red lighten-2"
              name="action"
            >
              Send
              <i className="material-icons right">send</i>
            </button>
          </div>
        </div>
      </div>
      </div>
    );
  };
}


export default connect(null, {forgotPasswordAction})(ForgotPassword);