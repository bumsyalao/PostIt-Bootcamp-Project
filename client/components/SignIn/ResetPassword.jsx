import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import NavigationBar from '../NavigationBar';
import { resetPasswordAction } from '../../actions/SignInAction';


/**
 *
 *
 * @class ResetPassword
 * @extends {Component}
 */
class ResetPassword extends Component {

  /**
   * Creates an instance of ResetPassword.
   * @param {any} props
   *
   * @memberOf ResetPassword
   */
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      confirmPassword: ''
    };
    this.onChange = this.onChange.bind(this);
    this.resetPassword = this.resetPassword.bind(this);
  }

  /**
   *
   *
   * @param {any} event
   *
   * @memberOf ResetPassword
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
   *
   *
   * @param {any} event
   *
   * @memberOf ResetPassword
   */
  resetPassword() {
    const password = this.state.password;
    const pathname = this.props.location.pathname;
    const hash = pathname.replace('/reset-password/', '');
    this.props.resetPasswordAction({ password, hash })
      .then(() => {
        this.props.history.push('/homepage/welcome-page');
        Materialize.toast('Password Updated Succesfully', 5000, 'green');
      })
      .catch(err => Materialize.toast(err.response.data.message, 5000, 'red'));
  }

  /**
   *
   *
   * @returns
   *
   * @memberOf ResetPassword
   */
  render() {
    return (
      <div>
      <NavigationBar match={this.props.match}/>
      <div className="form-margin">
        <div className="col s12 container">
          <div className="input-field col s6 offset-s3">
            <i className="material-icons prefix">lock</i>
            <input
            value={this.state.password}
            onChange={this.onChange}
            name="password"
            type="password"
            className="validate"
            required/>
            <label className="active" htmlFor="password">password</label>
          </div>
          <div className="input-field col s6 offset-s3">
            <i className="material-icons prefix">lock</i>
            <input
            value={this.state.confirmPassword}
            onChange={this.onChange}
            name="confirmPassword"
            type="password"
            className="validate"
            required/>
            <label className="active" htmlFor="confirmPassword">confirm password</label>
          </div>
          <div className="input-field col s6 offset-s3">
            <button
              onClick={this.resetPassword}
              className="btn waves-effect waves-light col s6 offset-s3 red lighten-2"
              name="action">
              Reset Password
              <i className="material-icons right">send</i>
            </button>
          </div>
        </div>
      </div>
      </div>
    );
  }
}
export default withRouter(
  connect(null, { resetPasswordAction })(ResetPassword));
