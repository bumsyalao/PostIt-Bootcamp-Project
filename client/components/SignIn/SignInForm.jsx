/* global Materialize */
import React from 'react';
import PropTypes from 'prop-types';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignInRequest } from '../../actions/SignInAction';

/**
 * @class SignInForm
 * @extends {React.Component}
 */
class SignInForm extends React.Component {
  /**
   * Creates an instance of SignInForm.
   * Binds class methods.
   * @param {any} props
   *
   * @memberOf SignInForm
   */
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
      username: '',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
    /**
   *
   * Check if user is authenticated
   * Directs user to homepage if user is already signed in
   *
   * @memberOf Homepage
   */
  componentDidMount() {
    if (this.props.access.isAuthenticated) {
      this.props.history.push('/homepage/welcome-page');
    }
  }
  /**
   * Sets the event value to the state
   * @return {void}
   * @param {Object} event The event of the HTML component
   *
   * @memberOf SignInForm
   */
  onChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }
  /**
   * Validates the SignInForm
   * Makes an action call to sign in the user
   * Toasts the error/success message
   * @return {void}
   *
   * @memberOf SignInForm
   */
  onSubmit() {
    const SigninInfo = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.userSignInRequest(SigninInfo)
      .then((res) => {
        if (this.props.access.message) {
          this.props.history.push('/homepage/welcome-page');
          Materialize.toast(res.message, 5000, 'green');
        } else {
          Materialize.toast(`Login Failed: ${res.response.data.message}`, 5000, 'red');
        }
      })
      .catch((error) => {
        Materialize.toast(error.response.data.message, 5000, 'red');
      });
  }

  /**
   *
   * Renders SignInForm component
   * @returns SignInForm HTML component
   *
   * @memberOf SignInForm
   */
  render() {
    return (
      <div className="form-margin">
      <div><h3 className="form-header">LOGIN</h3></div>
        <div className="col s12 container">
          <div className="input-field col s6 offset-s3">
            <i className="material-icons prefix">account_circle</i>
            <input
              id="username"
              value={this.state.username}
              onChange={this.onChange}
              name="username"
              type="text"
              className="validate"
              required
            />
            <label className="active" htmlFor="credential">
              username
            </label>
          </div>
          <div className="input-field col s6 offset-s3">
            <i className="material-icons prefix">lock</i>
            <input
              id="password"
              value={this.state.password}
              onChange={this.onChange}
              name="password"
              type="password"
              className="validate"
              required
            />
            <label className="active" htmlFor="password">
              password
            </label>
          </div>
          <button
            onClick={this.onSubmit}
            id="submit-signin"
            className=
            "btn waves-effect waves-light col s6 offset-s3 red lighten-2"
            name="action"
          >
            Login<i className="material-icons right">send</i>
          </button>
        </div>
        <div className="col s6 offset-s3 forgot-password">
          <Link to="/forgot-password">Forgot Password?</Link>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  access: state.access
});

SignInForm.propTypes = {
  userSignInRequest: PropTypes.func.isRequired
};

export default connect(mapStateToProps, { userSignInRequest })(
  withRouter(SignInForm)
);
