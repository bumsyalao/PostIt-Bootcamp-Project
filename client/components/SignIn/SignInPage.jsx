import React from 'react';
import SignInForm from './SignInForm';
import NavigationBar from '../NavigationBar';


/**
 * @class SignInPage
 * @extends {React.Component}
 */
const SignInPage = ({ match }) => (
    <div>
      <NavigationBar match={match}/>
    <div className="row">
      <div className="col-md-4 col-md-offset-4">
        <SignInForm />
      </div>
    </div>
    </div>
);

export default SignInPage;
