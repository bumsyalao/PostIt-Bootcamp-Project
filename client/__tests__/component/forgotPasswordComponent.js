/* global expect jest jest */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import { ForgotPassword } from '../../components/SignIn/ForgotPassword';

configure({ adapter: new adapter() });

describe('ForgotPassword component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    forgotPasswordAction: jest.fn(() => Promise.resolve()),
    history: {
      push: spy
    }
  };
  const component = shallow(
    <ForgotPassword {...props} />
  );
  it('should match the ForgotPassword snapshot', () => {
    const tree = toJson(component);
    expect(component.find('input').length).toEqual(1);
    expect(tree).toMatchSnapshot();
  });
  it('should set state when onChange function is called', () => {
    component.instance().onChange(
      { target: { value: 'banku@email.com', id: 'email' } });
    expect(component.state('email')).toEqual('banku@email.com');
  });

  it('forgotPassword function should run', () => {
    component.instance().onChange(
      { target: { value: 'banku@email.com', id: 'email' } });
    expect(component.state('email')).toEqual('banku@email.com');
    component.find('#submit-forgotPassword').simulate('click');
    expect(component.find('#submit-forgotPassword').length).toEqual(1);
  });
});
