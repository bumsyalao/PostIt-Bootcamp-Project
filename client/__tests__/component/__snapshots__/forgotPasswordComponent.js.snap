// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component should match the ForgotPassword snapshot 1`] = `
<div>
  <Connect(withRouter(NavigationBar)) />
  <div
    className="form-margin"
  >
    <div
      className="col s12 container"
    >
      <div
        className="input-field col s6 offset-s3"
      >
        <i
          className="material-icons prefix"
        >
          email
        </i>
        <input
          className="validate"
          id="email"
          name="email"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <label
          className="active"
          htmlFor="credential"
        >
          Type in your email address so we can send you instructions to reset your password
        </label>
      </div>
      <div
        className="input-field col s6 offset-s3"
      >
        <button
          className="btn waves-effect waves-light col s6 offset-s3 red lighten-2"
          id="submit-forgotPassword"
          name="action"
          onClick={[Function]}
        >
          Send
          <i
            className="material-icons right"
          >
            send
          </i>
        </button>
      </div>
    </div>
  </div>
</div>
`;
